pkg/distributor/query.go

```diff
@@ -232,10 +232,16 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, userID string, re
 					return nil, validation.LimitError(fmt.Sprintf(errMaxChunksPerQueryLimit, util.LabelMatchersToString(matchers), chunksLimit))
 				}
 			}
+
 			for _, series := range resp.Chunkseries {
 				if limitErr := queryLimiter.AddSeries(series.Labels); limitErr != nil {
 					return nil, limitErr
 				}
+				for _, chunks := range series.Chunks {
```

Comment:
You can add a function on `TimeSeriesChunk` (in `pkg/ingester/client/custom.go`) to count the total size of all chunks in `TimeSeriesChunk` and then call `queryLimiter.AddChunkBytes()` just once. This will simplify the logic here.